### PR TITLE
Remove code duplication.

### PR DESCRIPTION
**Pull request by danielfenner**
Cleanup from previous PR. Expose ExecuteMachineCode in header so we
don't need to duplicate it in the test.

Bug: N/A
Test: Existing unit tests.

### CODE REVIEW

File: src/UserSpaceInstrumentation/InjectLibraryInTracee.cpp [hunk 1]
```diff
@@ -326,4 +269,57 @@ ErrorMessageOr<uint64_t> FindFunctionAddress(pid_t pid, std::string_view functio
...
+  OUTCOME_TRY(original_registers.BackupRegisters(pid));
+
+  RegisterState registers_for_execution = original_registers;
+  registers_for_execution.GetGeneralPurposeRegisters()->x86_64.rip = address_code;
+  // The calling convention for x64 assumes the 128 bytes below rsp to be usable as a scratch pad
+  // for the current function. This area is called the 'red zone'. The function we interrupted might
+  // have stored temporary data in the red zone and the function we are about to execute might do
+  // the same. To keep them separated we decrement rsp by 128 bytes.
+  // The calling convention also asks for rsp being a multiple of 16 so we additionally round down
+  // to the next multiople of 16.
```
Comment (dimitry-) [inline src/UserSpaceInstrumentation/InjectLibraryInTracee.cpp:291]: nit: to be aligned to 16 bytes

**Review by dimitry- — APPROVED** [review]

File: src/UserSpaceInstrumentation/InjectLibraryInTracee.cpp (hunk 1 above)
Comment (pierricgimmig) [reply src/UserSpaceInstrumentation/InjectLibraryInTracee.cpp]: or "to the previous multiple"?

**Review by pierricgimmig — APPROVED** [review]
Cool, thanks!

File: src/UserSpaceInstrumentation/InjectLibraryInTracee.cpp (hunk 1 above)
Comment (danielfenner) [reply src/UserSpaceInstrumentation/InjectLibraryInTracee.cpp]: both done.

**Pull request merged**